### PR TITLE
Bypass DirLockGaurd in Options

### PR DESCRIPTION
**Pull request by ehsannm**
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1046)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,5 @@
 p/
 badger-test*/
+.idea*
+.idea/*
+.idea
```
[the remark this answers is not in the document]
Comment (ehsannm) [reply .gitignore]: they added to gitignore

**Comment by jon-tksl** [comment]
Is this dead, because it would be really, really useful for me.

**Comment by jonknight73** [comment]
Is this completely defunct, because it would be really. really useful for me?

**Comment by jarifibrahim** [comment]
@jonknight73 I've raised https://github.com/dgraph-io/badger/pull/1243 for this issue.

**Pull request closed without merging**